src/Altinn.Profile/Controllers/UsersController.cs

```diff
@@ -123,4 +127,55 @@ public async Task<ActionResult<UserProfile>> GetUserFromSSN([FromBody] string ss
             userProfile => Ok(userProfile),
             _ => NotFound());
     }
+
+    /// <summary>
+    /// Updates the profile settings of the current user based on the request context
+    /// </summary>
+    /// <returns>User profile of current user</returns>
+    [HttpPut("current/profilesettings")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<ProfileSettingPreference>> UpdateProfileSettings([FromBody]ProfileSettingPreference request)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        string userIdString = Request.HttpContext.User.Claims
+            .Where(c => c.Type == AltinnCoreClaimTypes.UserId)
+            .Select(c => c.Value).SingleOrDefault();
+
+        if (string.IsNullOrEmpty(userIdString))
+        {
+            return BadRequest("Invalid request context. UserId must be provided in claims.");
+        }
+
+        int userId = int.Parse(userIdString);
+
+        var profileSettings = new ProfileSettings
+        {
+            UserId = userId,
+            LanguageType = request.Language,
+            DoNotPromptForParty = request.DoNotPromptForParty,
+            PreselectedPartyUuid = request.PreselectedPartyUuid,
+            ShowClientUnits = request.ShowClientUnits,
+            ShouldShowSubEntities = request.ShouldShowSubEntities,
+            ShouldShowDeletedEntities = request.ShouldShowDeletedEntities
+        };
```

Comment:
What's the plan for IgnoreUnitProfileDateTime?

Reply:
I can make them nullable, the same as IgnoreUnitProfileDateTime
```
            // Only update IgnoreUnitProfileDateTime if the other instance has a value
            if (other.IgnoreUnitProfileDateTime.HasValue)
            {
                IgnoreUnitProfileDateTime = other.IgnoreUnitProfileDateTime;
            }
```